aten/src/ATen/native/TensorShape.cpp

```diff
@@ -596,6 +596,23 @@ Tensor & unsqueeze_(Tensor& self, int64_t dim) {
   return self.as_strided_(std::get<0>(g), std::get<1>(g));
 }
 
+Tensor flatten(const Tensor& self, int64_t start, int64_t end) {
+  start = maybe_wrap_dim(start, self.dim());
+  end = maybe_wrap_dim(end, self.dim());
+  AT_CHECK(start < end, "start dim must be before end dim");
+
+  std::vector<int64_t> shape;
+  for (int i = 0; i < start; i++) {
+    shape.push_back(self.sizes()[i]);
```

Comment:
Can we hoist `self.sizes()` above the loops and not construct a new `IntList` in every loop iteration (also below)

Reply:
or just use `self.size(i)`